common/src/main/java/io/seata/common/loader/EnhancedServiceLoader.java

```diff
@@ -198,181 +173,445 @@
+                                    LOGGER.warn("load [{}] class fail. {}", line, e.getMessage());
+                                }
                             }
                         }
+                    } catch (Throwable e) {
+                        LOGGER.warn(e.getMessage());
+                    } finally {
+                        IOUtil.close(reader);
                     }
-                } catch (Throwable e) {
-                    LOGGER.warn(e.getMessage());
-                } finally {
-                    IOUtil.close(reader);
                 }
             }
         }
-    }
 
-    /**
-     * init instance
-     *
-     * @param <S>       the type parameter
-     * @param service   the service
-     * @param implClazz the impl clazz
-     * @param argTypes  the arg types
-     * @param args      the args
-     * @return s s
-     * @throws IllegalAccessException the illegal access exception
-     * @throws InstantiationException the instantiation exception
-     * @throws NoSuchMethodException the no such method exception
-     * @throws InvocationTargetException the invocation target exception
-     */
-    protected static <S> S initInstance(Class<S> service, Class implClazz, Class[] argTypes, Object[] args)
-        throws IllegalAccessException, InstantiationException, NoSuchMethodException, InvocationTargetException {
-        S s = null;
-        if (argTypes != null && args != null) {
-            // Constructor with arguments
-            Constructor<S> constructor = implClazz.getDeclaredConstructor(argTypes);
-            s = service.cast(constructor.newInstance(args));
-        } else {
-            // default Constructor
-            s = service.cast(implClazz.newInstance());
+        private ExtensionDefinition getExtensionDefinition(Class<?> clazz) {
+            String serviceName = null;
+            Integer priority = 0;
+            Scope scope = Scope.SINGLETON;
+            LoadLevel loadLevel = clazz.getAnnotation(LoadLevel.class);
+            if (loadLevel != null) {
+                serviceName = loadLevel.name();
+                priority = loadLevel.order();
+                scope = loadLevel.scope();
+            }
+            ExtensionDefinition result = new ExtensionDefinition(serviceName, priority, scope, clazz);
+            //do cache
+            classToDefinitionMap.put(clazz, result);
+            if (serviceName != null) {
+                if (nameToDefinitionsMap.containsKey(serviceName)) {
+                    List<ExtensionDefinition> definitions = nameToDefinitionsMap.get(serviceName);
+                    definitions.add(result);
+                } else {
+                    List<ExtensionDefinition> definitions = new ArrayList<>();
+                    definitions.add(result);
+                    nameToDefinitionsMap.put(serviceName, definitions);
+                }
+            }
+            if (defaultExtensionDefinition == null || priority >= highestLoadPriority) {
```

Comment:
As said before, this statement should be `if (defaultExtensionDefinition == null || priority > highestLoadPriority)`.

Reply:
This part of code was removed cause that they wer unnecessary.